corehq/apps/app_manager/fixtures/mobile_ucr.py

```diff
@@ -163,6 +202,59 @@ def report_config_to_fixture(report_config, restore_user):
         return report_elem
 
     @staticmethod
+    def report_config_to_v2_fixture(report_config, restore_user):
+        domain = restore_user.domain
+        report, data_source = ReportFixturesProvider._get_report_and_data_source(
+            report_config.report_id, domain
+        )
+        report_fixture_id = 'commcare-reports:' + report_config.report_id
+
+        # TODO: Convert to be compatible with restore_user
+        # apply filters specified in report module
+        all_filter_values = {
+            filter_slug: restore_user.get_ucr_filter_value(filter, report.get_ui_filter(filter_slug))
+            for filter_slug, filter in report_config.filters.items()
+        }
+        # apply all prefilters
+        prefilters = [ReportFilterFactory.from_spec(p, report) for p in report.prefilters]
+        prefilter_values = {prefilter.name: prefilter.value() for prefilter in prefilters}
+        all_filter_values.update(prefilter_values)
+        # filter out nulls
+        filter_values = {
+            filter_slug: filter_value for filter_slug, filter_value in all_filter_values.items()
+            if filter_value is not None
+        }
+        defer_filters = {
+            filter_slug: report.get_ui_filter(filter_slug)
+            for filter_slug, filter_value in all_filter_values.items()
+            if filter_value is None and is_valid_mobile_select_filter_type(report.get_ui_filter(filter_slug))
+        }
+        data_source.set_filter_values(filter_values)
+        data_source.defer_filters(defer_filters)
+        filter_options_by_field = defaultdict(set)
+
+        rows_elem = ReportFixturesProvider._get_v2_report_elem(
+            data_source,
+            {ui_filter.field for ui_filter in defer_filters.values()},
+            filter_options_by_field
+        )
+        filters_elem = ReportFixturesProvider._get_filters_elem(
+            defer_filters, filter_options_by_field, restore_user._couch_user)
+        filters_elem.attrib['id'] = report_fixture_id
+
+        if (data_source.config.backend_id in UCR_SUPPORT_BOTH_BACKENDS and
+                random.randint(0, MOBILE_UCR_RANDOM_THRESHOLD) == MOBILE_UCR_RANDOM_THRESHOLD):
+            compare_ucr_dbs.delay(domain, report_config.report_id, filter_values)
+
+        report_elem = E.fixture(
+            id=report_fixture_id, user_id=restore_user.user_id,
+            report_id=report_config.report_id, last_sync=_utcnow().isoformat(),
+            indexed='true'
+        )
+        report_elem.append(rows_elem)
+        return [filters_elem, report_elem]
```

Comment:
Lots of duplicated code with v1

Reply:
Agreed. When initially writing this I copied it over, wrote a test and worked through differences. I can take a pass on being more DRY